docs/LOGGING.md

```diff
@@ -30,28 +30,49 @@ Logs coming from xen-tools container are of three kinds:
   * ```qemu-dm-[VM_NAME]``` logs the qemu device model output
   * ```qdisk-[VM ID]``` logs the qdisk output
 
-If there are too many logs coming from various containers and the log reader is not able to keep up, oldest logs from the current circular buffer will be overwritten. All logs read by memlgod by rsyslogd will be written to disk queues. When there is heavy disk usage and rsyslogd does not get sufficient time to write incoming logs to disk, it can result in log loss from memlogd.
+All logs from memlgod and from /dev/kmsg read by newlogd will be written to disk log file and then to be compressed into gzip log files. When there is heavy disk usage or CPU load and newlogd does not get sufficient time to write incoming logs to disk, it can result in log loss from memlogd.
 
-The following diagram shows the flow of logs from containers to rsyslog and to cloud.
+The following diagram shows the flow of logs from containers to newlogd and to cloud.
 
-![EVE Log Flow](pics/eve_log_flow.png)
+![EVE Log Flow](pics/eve_newlog_flow.png)
 
-## Log Aggregation
+## Log Aggregation, Reformating and Compression for Persistent Log Files
 
-All logs collected from various containers/services in the system will reach rsyslogd demon. Rsyslogd queues the incoming logs on disk. Rsyslogd is configured to use disk queues in order to minimize log loss during power/network failures.
+All logs collected from various containers/services/kernel in the system will reach newlogd deamon. Newlogd formats the log entries and writes into temporary log files on disk in /persist/newlog/collect directory. The logs from device Dom0 side will be saved to file with name prefix with 'dev_log.' and logs from DomU side will be saved to files with name prefix with 'app.APP-UUID.log' where APP-UUID is the application UUID assigned to DomU application. The temporary file is kept on the disk until either the file size has exceeded 400 KBytes or the elasped time on the file has been opened for longer than 5 minutes.
 
-Useful information about the current state of logs/queues can be seen in the following directories:
+When the above log file is closed either due to size or time limit has reached, it will be mvoed and compressed with gzip protocol into either 'devUpload' or 'appUpload' directory. The size of the gzip file is limited to 50 KBytes due to the northbound queueing configuration. If the compressed file is larger than the limit, it will be splitted and compressed into two separate gzip files. The gzip filename is encoded with current timestamp in Unix seconds, such as 'dev_log.1600831551.gz' for device log, and with timestamp and application UUID such as 'app.62195aa9-7db4-4ac0-86d3-d8abe0ff0ea9.log.1599186248' for application logs. The metadata such as device-UUID, the image partition, and image version or app Name for application are encoded as part of the gzip metadata header along with the gzip file.
```

Comment:
Typo "mvoed"

Reply:
Typo "splitted" -> "split"

Reply:
Shouldn't app.62195aa9-7db4-4ac0-86d3-d8abe0ff0ea9.log.1599186248 have a .gz at the end as well?

Reply:
What is the "gzip metadata header"? Something you are adding inside the beginning of each file?

Reply:
right. I missed a 'gz' in the app case.
the "gzip metadata header", is something can be encoded into the gz file, it's not part of the content of the file. In Go code, after getting the reader from gzip function call, this reader of gzip will have a reader.header structure, which includes timestamp in go time.Time format, and some 'Name' and 'Comment' string fields. Currently I use the all three, 'Name' for application-Name, and 'comment' for the json device-uuid, partition, etc info.

Reply:
Ah - I didn't know gzip had such a header. Thanks.